src/slashdbclient.js

```diff
@@ -58,52 +93,145 @@ class SlashDBClient {
     this.queryDefEP = '/querydef';  
               
   }
-
   /**
-   * Logs in to SlashDB instance.  Only required when using password-based login.
+   * Logs in to SlashDB instance.  Only required when using username/password based crendentials. if not provided will try SSO login.
+   * 
+   * @param {string} username - optional username to use when connecting to SlashDB instance using password based login
+   * @param {string} password - optional password associated with username
    * @returns {true} true - on successful login
    * @throws {Error} on invalid login or error in login process
    */
   async login(username, password) {
 
-    if (typeof(username) !== 'string') {
-      throw TypeError(SDB_SDBC_INVALID_USERNAME);
+    let body = {};
+    let sso = this.sso;
+    
+    // try {
+      if (password) {
+        body = { login: username, password: password };
+        let response = (await this.sdbConfig.post(body, this.loginEP)).res;
+        
+        if (response.ok === true) {
+          this.basic = btoa(username + ":" + password);
```

Comment:
Can you explain this a bit?  I think what's happening here is that if a username and password are provided, once login is completed, a Basic header with b64 contens is used for future requests...is that right?

Reply:
Yes, I was trying to perform the following [SlashDB basic authentication](https://docs.slashdb.com/user-guide/security/authentication/#basic-authentication), after a successful login, to keep an encoded version for authenticate other requests